src/components/imageoptionseditor/imageoptionseditor.js

```diff
@@ -1,77 +1,112 @@
-        })[0]
+        return imageOptions.filter(function (i) {
+            return i.Type == type;
+        })[0];
     }
 
     function getImageConfig(options, availableOptions, imageType, itemType) {
-        return findImageOptions(options.ImageOptions || [], imageType) || findImageOptions(availableOptions.DefaultImageOptions || [], imageType) || getDefaultImageConfig(itemType, imageType)
+        return findImageOptions(options.ImageOptions || [], imageType) || findImageOptions(availableOptions.DefaultImageOptions || [], imageType) || getDefaultImageConfig(itemType, imageType);
     }
 
     function setVisibilityOfBackdrops(elem, visible) {
-        visible ? (elem.classList.remove("hide"), elem.querySelector("input").setAttribute("required", "required")) : (elem.classList.add("hide"), elem.querySelector("input").setAttribute("required", ""), elem.querySelector("input").removeAttribute("required"))
+        if (visible) {
+            elem.classList.remove("hide");
+            elem.querySelector("input").setAttribute("required", "required");
+        } else {
+            elem.classList.add("hide");
+            elem.querySelector("input").setAttribute("required", "");
+            elem.querySelector("input").removeAttribute("required");
+        }
     }
 
     function loadValues(context, itemType, options, availableOptions) {
         var supportedImageTypes = availableOptions.SupportedImageTypes || [];
-        setVisibilityOfBackdrops(context.querySelector(".backdropFields"), -1 != supportedImageTypes.indexOf("Backdrop")), setVisibilityOfBackdrops(context.querySelector(".screenshotFields"), -1 != supportedImageTypes.indexOf("Screenshot")), Array.prototype.forEach.call(context.querySelectorAll(".imageType"), function(i) {
-            var imageType = i.getAttribute("data-imagetype"),
-                container = dom.parentWithTag(i, "LABEL"); - 1 == supportedImageTypes.indexOf(imageType) ? container.classList.add("hide") : container.classList.remove("hide"), getImageConfig(options, availableOptions, imageType, itemType).Limit ? i.checked = !0 : i.checked = !1
+        setVisibilityOfBackdrops(context.querySelector(".backdropFields"), -1 != supportedImageTypes.indexOf("Backdrop"));
+        setVisibilityOfBackdrops(context.querySelector(".screenshotFields"), -1 != supportedImageTypes.indexOf("Screenshot"));
+        Array.prototype.forEach.call(context.querySelectorAll(".imageType"), function (i) {
+            var imageType = i.getAttribute("data-imagetype");
+            var container = dom.parentWithTag(i, "LABEL");
+
+            if (-1 == supportedImageTypes.indexOf(imageType)) {
+                container.classList.add("hide");
+            } else {
+                container.classList.remove("hide");
+            }
+
+            if (getImageConfig(options, availableOptions, imageType, itemType).Limit) {
+                i.checked = true;
+            } else {
+                i.checked = false;
+            }
```

Comment:
maybe
```js
i.checked = !!(getImageConfig(options, availableOptions, imageType, itemType).Limit);
```
?

Reply:
why?

Reply:
because this is essentially what that 5-line `if` does - converts `getImageConfig()` to a boolean. `!!` does the same thing but shorter :)

Reply:
Though I don't insist on that change, if you don't like that ignore it :)